assembl/static2/js/app/components/administration/landingPage/customizeHeader.jsx

```diff
@@ -1,137 +1,174 @@
 import Helper from '../../common/helper';
-import {
-  updateLandingPageHeaderTitle,
-  updateLandingPageHeaderSubtitle,
-  updateLandingPageHeaderButtonLabel,
-  updateLandingPageHeaderImage,
-  updateLandingPageHeaderLogo
-} from '../../../actions/adminActions/landingPage';
+import { load, postLoadFormat } from './header/load';
+import { save, createMutationsPromises } from './header/save';
+import LoadSaveReinitializeForm from '../../form/LoadSaveReinitializeForm';
+import FileUploaderFieldAdapter from '../../form/fileUploaderFieldAdapter';
+import DatePickerFieldAdapter from '../../form/datePickerFieldAdapter';
+import MultilingualTextFieldAdapter from '../../form/multilingualTextFieldAdapter';
+import MultilingualRichTextFieldAdapter from '../../form/multilingualRichTextFieldAdapter';
+import AdminForm from '../../form/adminForm';
+import Loader from '../../common/loader';
+import { type DateTime } from '../../form/types.flow';
+import { validStartDate, validEndDate, validateDatePicker } from './header/validate';
+import DiscussionQuery from '../../../graphql/DiscussionQuery.graphql';
+import { convertISO8601StringToDate } from '../../../utils/globalFunctions';
+import manageErrorAndLoading from '../../common/manageErrorAndLoading';
 
 type Props = {
-  header: {
-    title: string,
-    subtitle: ?string,
-    buttonLabel: string,
-    headerImgMimeType: string,
-    headerImgUrl: string,
-    headerImgTitle: string,
-    logoImgMimeType: string,
-    logoImgUrl: string,
-    logoImgTitle: string
-  },
+  client: ApolloClient,
   editLocale: string,
-  handleTitleChange: Function,
-  handleSubtitleChange: Function,
-  handleButtonLabelChange: Function,
-  handleImageChange: Function,
-  handleLogoChange: Function
+  data: {
+    discussion: {
+      startDate: string,
+      endDate: string
+    }
+  }
 };
 
-export const DumbCustomizeHeader = ({
-  header,
-  handleTitleChange,
-  handleSubtitleChange,
-  handleButtonLabelChange,
-  handleImageChange,
-  handleLogoChange,
-  editLocale
-}: Props) => {
-  const {
-    title,
-    subtitle,
-    buttonLabel,
-    headerImgMimeType,
-    headerImgUrl,
-    headerImgTitle,
-    logoImgMimeType,
-    logoImgUrl,
-    logoImgTitle
-  } = header;
-  const titlePh = I18n.t('administration.landingPage.header.titleLabel');
-  const subtitlePh = I18n.t('administration.landingPage.header.subtitleLabel');
-  const buttonLabelPh = I18n.t('administration.landingPage.header.buttonLabel');
+type State = {
+  startDate: DateTime,
+  endDate: DateTime,
+  startDateConflict: boolean,
+  endDateConflict: boolean
+};
+
+const loading = <Loader />;
+
+export const DumbCustomizeHeader = ({ client, editLocale, data: { discussion: { startDate, endDate } } }: Props) => {
+  const state: State = {
+    startDate: convertISO8601StringToDate(startDate),
+    endDate: convertISO8601StringToDate(endDate),
+    startDateConflict: false,
+    endDateConflict: false
+  };
+
+  const onStartChange = (e: DateTime): void => {
```

Comment:
You should check dates in `validate.js`

Reply:
In this form, two `Field`s' pristine state depend on one another. Hence, I have to create local state, and pass these onChange handlers which will compare and validate between the two fields. At the core of the callbacks, the "check logic", is imported from validate.js.